digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetFiller.scala

```diff
@@ -42,27 +42,32 @@ object PointAssetFiller {
     }
   }
 
-   def correctShortened(asset: PersistedPointAsset , roadLinks: Seq[RoadLink], changeInfo: ChangeInfo, adjustmentOption: Option[AssetAdjustment]): Option[AssetAdjustment]={
-     roadLinks.find(_.linkId == changeInfo.newId.getOrElse(0L)) match {
-       case Some(newRoadLink) =>
-         val mValue = adjustmentOption match {
-           case Some (adjustment) =>
+  def correctShortened(asset: PersistedPointAsset , roadLinks: Seq[RoadLink], changeInfo: ChangeInfo, adjustmentOption: Option[AssetAdjustment]): Option[AssetAdjustment]= {
+    roadLinks.find(_.linkId == changeInfo.newId.getOrElse(0L)) match {
+      case Some(newRoadLink) =>
+        val mValue = adjustmentOption match {
+          case Some(adjustment) =>
             adjustment.mValue
-           case _ =>
+          case _ =>
             asset.mValue
-         }
-
-         if (changeInfo.oldStartMeasure.getOrElse (0.0) - MaxDistanceDiffAllowed <= mValue && changeInfo.oldStartMeasure.getOrElse (0.0) >= mValue)
-          correctGeometry (asset.id, newRoadLink, changeInfo.newStartMeasure.getOrElse (0.0), changeInfo.vvhTimeStamp)
-         else if (changeInfo.oldEndMeasure.getOrElse (0.0) + MaxDistanceDiffAllowed >= mValue && changeInfo.oldEndMeasure.getOrElse (0.0) <= mValue)
-          correctGeometry (asset.id, newRoadLink, changeInfo.newEndMeasure.getOrElse (0.0), changeInfo.vvhTimeStamp)
-         else
-          correctValuesAndGeometry (asset, roadLinks, changeInfo, adjustmentOption)
-       case _ =>
-         None
-     }
+        }
+        val assetpoint = Point(asset.lon, asset.lat)
+        val roadOption = GeometryUtils.calculatePointFromLinearReference(newRoadLink.geometry, mValue)
```

Comment:
The variables assetpoint and roadOption can be removed.
I think is not used in the code.